### PR TITLE
LSKNet implementation

### PR DESCRIPTION
**Pull request by zcablii**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

An implementation of LSKNet ("Large Selective Kernel Network for Remote Sensing Object Detection" at: https://arxiv.org/pdf/2303.09030.pdf)

## Modification

An implementation of LSKNet ("Large Selective Kernel Network for Remote Sensing Object Detection" at: https://arxiv.org/pdf/2303.09030.pdf)

## BC-breaking (Optional)

Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.
N/A

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.
N/A

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by RangiLyu** [comment]
Thanks for your contribution! Before merge this PR, you need to format your code with pre-commit. Please run `pip install pre-commit` and then run `pre-commit run --all-files` to format your code.

File: projects/LSKNet/README.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+| [RTMDet-l](https://arxiv.org/abs/2212.07784) | 90.60  |  97.10  | le90  |   3x    |   -    |     -   |       -    | Prev. Best |
+| [ReDet](https://arxiv.org/abs/2103.07733) | 90.46  |  97.63  | le90  |   3x    |   2*4    |            [redet_re50_refpn_3x_hrsc_le90](./configs/redet/redet_re50_refpn_3x_hrsc_le90.py)             |       -    | Prev. Best |
+| LSKNet_S | 90.65  |  98.46  | le90  |   3x    |    1*8     |            [lsk_s_fpn_3x_hrsc_le90](./configs/lsknet/lsk_s_fpn_3x_hrsc_le90.py)             |         [model](https://pan.baidu.com/s/1jRLb5m6tGab6BF1ky1JP6A?pwd=bosr) \| [log](https://pan.baidu.com/s/1f0i5oGn3QseKQLAGMcy4_w?pwd=kn0x)         | |
```
Comment (RangiLyu) [inline projects/LSKNet/README.md:55]: Would you like to allow us to host your models on our server?

Comment (zcablii) [reply projects/LSKNet/README.md:55]: Sure! That will be great!

File: projects/LSKNet/README.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+LSKNet backbone code is placed under mmrotate/models/backbones/, and the train/test configure files are placed under configs/lsknet/
+
+## Usage
+
+### Training commands
+
+In MMRotate's root directory, run the following command to train the model:
+
+```bash
+python tools/train.py projects/example_project/configs/lsk_t_fpn_1x_dota_le90.py
```
Comment (zytx121) [inline projects/LSKNet/README.md:24]:
```suggestion
python tools/train.py projects/LSKNet/configs/lsk_t_fpn_1x_dota_le90.py
```

File: projects/LSKNet/README.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+## Description
+
+Author: @Yuxuan Li.
+This project is an implementation of "Large Selective Kernel Network for Remote Sensing Object Detection" at: [https://arxiv.org/pdf/2303.09030.pdf](https://arxiv.org/pdf/2303.09030.pdf)
+
+LSKNet backbone code is placed under mmrotate/models/backbones/, and the train/test configure files are placed under configs/lsknet/
```
Comment (zytx121) [inline projects/LSKNet/README.md:15]:
```suggestion
```

File: projects/LSKNet/README.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
+[![PWC](https://img.shields.io/endpoint.svg?url=https://paperswithcode.com/badge/large-selective-kernel-network-for-remote/object-detection-in-aerial-images-on-dota-1)](https://paperswithcode.com/sota/object-detection-in-aerial-images-on-dota-1?p=large-selective-kernel-network-for-remote)
```
Comment (zytx121) [inline projects/LSKNet/README.md:1]:
```suggestion
# LSKNet

[![PWC](https://img.shields.io/endpoint.svg?url=https://paperswithcode.com/badge/large-selective-kernel-network-for-remote/object-detection-in-aerial-images-on-dota-1)](https://paperswithcode.com/sota/object-detection-in-aerial-images-on-dota-1?p=large-selective-kernel-network-for-remote)
```

File: projects/LSKNet/configs/lsk_s_ema_fpn_1x_dota_le90.py [hunk 5]
```diff
@@ -0,0 +1,154 @@
+_base_ = [
+    '../../../configs/_base_/datasets/dota_ms.py',
+    '../../../configs/_base_/schedules/schedule_1x.py',
+    '../../../configs/_base_/default_runtime.py'
```
Comment (liuyanyi) [inline projects/LSKNet/configs/lsk_s_ema_fpn_1x_dota_le90.py:4]: Here we can use `mmrotate::` to replace these dots

Example
```
'mmrotate::_base_/datasets/dota_ms.py'
```

File: projects/LSKNet/configs/lsk_s_ema_fpn_1x_dota_le90.py [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+optim_wrapper = dict(
+    optimizer=dict(
+        _delete_=True,
+        type='AdamW',
+        lr=0.0002,
+        betas=(0.9, 0.999),
+        weight_decay=0.05))
+
+custom_hooks = [
+    dict(type='ExpMomentumEMAHook', total_iter=8541 * 12, priority=49)
```
Comment (liuyanyi) [inline projects/LSKNet/configs/lsk_s_ema_fpn_1x_dota_le90.py:153]: It seems `ExpMomentumEMAHook` has been replaced by `EMAHook` in mmengine? I'm not sure.

File: projects/LSKNet/configs/lsk_s_ema_fpn_1x_dota_le90.py (hunk 5 above)
Comment (zytx121) [reply projects/LSKNet/configs/lsk_s_ema_fpn_1x_dota_le90.py]: Hi @zcablii, this is a syntax sugar brought with OpenMMLab 2.0, which can easily which can easily inherit configuration files from different MM libraries.

https://github.com/open-mmlab/mmdetection/blob/dev-3.x/projects/EfficientDet/configs/efficientdet_effb0_bifpn_8xb16-crop512-300e_coco.py

**Review by zytx121 — APPROVED** [review]
LGTM. Just wait for the model link to be updated.

File: projects/LSKNet/README.md [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+
+## Results
+
+Imagenet 300-epoch pre-trained LSKNet-T backbone: [Download](https://pan.baidu.com/s/1CQHQBLR9UVW-7cxg7LB2Dg?pwd=84jd)
+
+Imagenet 300-epoch pre-trained LSKNet-S backbone: [Download](https://pan.baidu.com/s/1o-Pq8k_7qcTgfTWiBve6uA?pwd=dm3w)
+
+DOTA1.0
+
+|                           Model                            |  mAP  | Angle | lr schd | Batch Size |                                   Configs                                    |                                                               Download                                                               |     note     |
```
Comment (RangiLyu) [inline projects/LSKNet/README.md:41]: Hi, I uploaded the models and logs to our server. You can update the links in these tables.

pretrained models:
https://download.openmmlab.com/mmrotate/v1.0/lsknet/backbones/lsk_s_backbone-e9d2e551.pth
https://download.openmmlab.com/mmrotate/v1.0/lsknet/backbones/lsk_t_backbone-2ef8a593.pth

lsk_s_fpn_1x_dota_le90:
https://download.openmmlab.com/mmrotate/v1.0/lsknet/lsk_s_fpn_1x_dota_le90/lsk_t_fpn_1x_dota_le90_20230116.log
https://download.openmmlab.com/mmrotate/v1.0/lsknet/lsk_s_fpn_1x_dota_le90/lsk_t_fpn_1x_dota_le90_20230116-99749191.pth

lsk_t_fpn_1x_dota_le90:
https://download.openmmlab.com/mmrotate/v1.0/lsknet/lsk_t_fpn_1x_dota_le90/lsk_t_fpn_1x_dota_le90_20230206.log
https://download.openmmlab.com/mmrotate/v1.0/lsknet/lsk_t_fpn_1x_dota_le90/lsk_t_fpn_1x_dota_le90_20230206-3ccee254.pth

lsk_s_ema_fpn_1x_dota_le90:
https://download.openmmlab.com/mmrotate/v1.0/lsknet/lsk_s_ema_fpn_1x_dota_le90/lsk_s_ema_fpn_1x_dota_le90_20230212.log
https://download.openmmlab.com/mmrotate/v1.0/lsknet/lsk_s_ema_fpn_1x_dota_le90/lsk_s_ema_fpn_1x_dota_le90_20230212-30ed4041.pth

lsk_s_fpn_3x_hrsc_le90:
https://download.openmmlab.com/mmrotate/v1.0/lsknet/lsk_s_fpn_3x_hrsc_le90/lsk_s_fpn_3x_hrsc_le90_20230205.log
https://download.openmmlab.com/mmrotate/v1.0/lsknet/lsk_s_fpn_3x_hrsc_le90/lsk_s_fpn_3x_hrsc_le90_20230205-4a4a39ce.pth

File: projects/LSKNet/configs/lsk_s_ema_fpn_1x_dota_le90.py [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+        boxtype2tensor=False),
+    backbone=dict(
+        type='LSKNet',
+        embed_dims=[64, 128, 320, 512],
+        drop_rate=0.1,
+        drop_path_rate=0.1,
+        depths=[2, 2, 4, 2],
+        init_cfg=dict(
+            type='Pretrained',
+            checkpoint='data/pretrained/lsk_s_backbone.pth.tar'),
```
Comment (RangiLyu) [inline projects/LSKNet/configs/lsk_s_ema_fpn_1x_dota_le90.py:27]: update this link

**Review by RangiLyu — APPROVED** [review]

**Comment by OpenMMLab-Assistant-004** [comment]
Hi @zcablii ！First of all, we want to express our gratitude for your significant PR in the mmrotate project. Your contribution is highly appreciated, and we are grateful for your efforts in helping improve this open-source project during your personal time. We believe that many developers will benefit from your PR.

We would also like to invite you to join our Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/raweFPmdzG

If you have WeChat，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤

**Pull request merged**